Add MCP Documentation (#683)
eclipse-theia/theia-website: src/docs/user_ai.md

Custom agents can be configured in the AI Configuration View just like other chat agents. You can enable/disable them, modify their prompt templates, and integrate variables and functions within these templates to enhance functionality.

> Here is the updated **MCP Integration** section with the requested changes:
> 

Reviewer: ```suggestion
```

## MCP Integration

> The Theia IDE now supports an with the Model Context Protocol (MCP), enabling users to configure and utilize external services in their AI workflows. 

Reviewer: ```suggestion
The Theia IDE supports the Model Context Protocol (MCP), enabling users to configure and utilize external services in their AI workflows. 
```

> *Please note: While this integration does not yet include MCP servers in any standard prompts, it already allows end users to explore the MCP ecosystem and discover interesting new use cases. In the future, we plan to provide ready-to-use prompts using MCP servers and support auto-starting configured servers.*

Reviewer: ```suggestion
*Please note: While this integration does not yet include MCP servers in any standard prompts, it already allows end users to integrate MCP servers, explore the MCP ecosysten, and discover interesting new use cases. In the future, we plan to provide off-the-shelf prompts using MCP servers and support starting configured servers automatically.*
```

For a list of available MCP servers, visit the [MCP Servers Repository](https://github.com/modelcontextprotocol/servers).

### Configuring MCP Servers

To configure MCP servers, open the preferences and add entries to the `MCP Servers Configuration` section. Each server requires a unique identifier (e.g., `"brave-search"` or `"filesystem"`) and configuration details such as the command, arguments, and optional environment variables.

**Example Configuration:**

```json
{
  "brave-search": {
    "command": "npx",
    "args": [
      "-y",
      "@modelcontextprotocol/server-brave-search"
    ],
    "env": {
      "BRAVE_API_KEY": "YOUR_API_KEY"
    }
  },
  "filesystem": {
    "command": "npx",
    "args": ["-y", "@modelcontextprotocol/server-filesystem", "/Users/YOUR_USERNAME/Desktop"],
    "env": {
      "CUSTOM_ENV_VAR": "custom-value"
    }
  }
}
```

The configuration options include:
- **`command`**: The executable used to start the server (e.g., `npx`).
- **`args`**: An array of arguments passed to the command.
- **`env`**: An optional set of environment variables for the server.

> 

Reviewer 2: I suggest adding the following text here (don't know how to suggest quoted text): 

Note that on Windows, you need to start a command interpreter (e.g. `cmd.exe`) as the server command in order for path lookups to work as expected. The effective command line is then passed as an argument. For example:

```
"filesystem": {
    "command": "cmd",
    "args": ["/C", "npx -y @modelcontextprotocol/server-filesystem /Users/YOUR_USERNAME/Desktop"],
    "env": {
      "CUSTOM_ENV_VAR": "custom-value"
    }
  }
```
